airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -197,10 +209,15 @@ def __init__(self):
         self._validate()
 
     def _validate(self):
-        if not self.dags_volume_claim and (not self.git_repo or not self.git_branch):
+        # TODO: use XOR for dags_volume_claim and git_dags_folder_mount_point
+        if not self.dags_volume_claim \
+           and (not self.git_repo or not self.git_branch
+                or not self.git_dags_folder_mount_point) \
+           and not self.dags_volume_host:
```

Comment:
Can this condition look like the error message? e.g.

```python
if not self.dags_volume_claim \
   and not self.dags_volume_host \
   and not (self.git_repo and self.git_branch and self.git_dags_folder_mount_point):
```

Reply:
fixed